src/helpers.c

```diff
@@ -1180,3 +1180,33 @@ bool check_mix_exists(void)
 	free_string(&fullpath);
 	return ret;
 }
+
+/* Iterate the file list and remove from the file system each file/directory */
+void remove_files_from_fs(struct list *files)
+{
+	struct list *iter = NULL;
+	struct file *file = NULL;
+	char *fullfile = NULL;
+	int total = list_len(files);
+	int deleted = total;
+	int count = 0;
+
+	iter = list_head(files);
+	while (iter) {
+		file = iter->data;
+		iter = iter->next;
+		string_or_die(&fullfile, "%s/%s", globals.path_prefix, file->filename);
+		if (swupd_rm(fullfile) == -1) {
+			/* if a -1 is returned it means there was an issue deleting the
+			 * file or directory, in that case decrease the counter of deleted
+			 * files.
+			 * Note: If a file didn't exist it will still be counted as deleted,
```

Comment:
Can't you check for that? Why do we have this limitation?
Is there a problem on using the following?

    if (swupd_rm(fullfile) < 0) {

Reply:
I'm not sure, that part of the code (and the comment) already existed so I just copy/pasted it

Reply:
I just assumed it was true

Reply:
oh, got it. I though it was a new code. you can leave the way it is then